etl/pipelines/arkg_builder_pipeline.py

```diff
@@ -0,0 +1,125 @@
+from pathlib import Path
+
+from pyoxigraph import Literal, NamedNode, Quad, Store
+from requests_cache import CachedSession
+
+from etl.models.types import AntiRecommendationKey, RecordKey
+from etl.namespaces import ARKG, RDF, SCHEMA, WD
+
+
+class ArkgBuilderPipeline:
+    """
+    A pipeline to build Anti-Recommendation Knowledge Graphs.
+
+    Constructs a RDF Store from a tuple of anti-recommendation graphs.
+    """
+
+    def __init__(self, requests_cache_directory: Path) -> None:
+        self.__requests_cache_directory = requests_cache_directory
+        self.__store = Store()
+
+    def __add_anti_recommendation_quads_to_store(
+        self,
+        *,
+        anti_recommendation_keys: tuple[AntiRecommendationKey, ...],
+        item_reviewed: NamedNode,
+    ) -> None:
+        """
+        Add `anti-recommendation` Quads to the RDF Store.
+
+        `item_reviewed` is the IRI of the item that is being anti-recommended.
+
+        Each anti_recommendation_key has 2 Quad expressions in this method:
+        - a `type` Quad that expresses the type of Review the anti_recommendation_key belongs to.
+        - an `item-reviewed` Quad that relates the anti_recommendation_key to the item that is being anti-recommended.
+        """
+        for anti_recommendation_key in anti_recommendation_keys:
+
+            anti_recommendation_iri = ARKG.anti_recommendation_iri(
+                record_key=anti_recommendation_key
+            )
+
+            self.__store.add(
+                Quad(
+                    anti_recommendation_iri,
+                    RDF.TYPE,
+                    SCHEMA.RECOMMENDATION,
+                )
+            )
+            self.__store.add(
+                Quad(
+                    anti_recommendation_iri,
+                    SCHEMA.ITEMREVIEWED,
+                    item_reviewed,
+                )
+            )
+
+    def __get_wikidata_iri(self, record_key: RecordKey) -> NamedNode:
+        """Return a RDF node that contains the Wikidata IRI of record_key."""
+
+        self.__requests_cache_directory.mkdir(parents=True, exist_ok=True)
+
+        session = CachedSession(
```

Comment:
Construct this in the pipeline constructor?

Reply:
Yes, `cached_session` is now constructed in the constructor.